src/core/castCylinder.lua

```diff
@@ -8,22 +8,21 @@ local CAST_COUNT = 32
 -- http://www.ilikebigbits.com/blog/2015/3/2/plane-from-points
 local function planeFromPoints(points, weights)
 	if #points < 3 then
-		-- at least three points required
 		return
 	end
 
-	local n = 0
-	local sumX, sumY, sumZ = 0, 0, 0
-	for i, p in pairs(points) do
-		local w = weights[i]
-		n = n + w
-		sumX, sumY, sumZ = sumX + p.X*w, sumY + p.Y*w, sumZ + p.Z*w
+	local pointSum = Vector3.new()
+	local weightSum = 0
+	for i, point in pairs(points) do
+		local weight = weights[i]
+		weightSum = weightSum + weight
+		pointSum = pointSum + point*weight
```

Comment:
This allocates 2 new vectors for each loop. That's an additional 64 trash vector allocs per luanoid per heatbeat.

Might be premature to worry about garbage allocs, but... thoughts?